keps/sig-release/3720-freezing-k8s-gcr-io/README.md

```diff
@@ -0,0 +1,158 @@
+  - [X] (N/A) Minimum Two Week Window for GA e2e tests to prove flake free
+- [X] (N/A) Graduation criteria is in place
+  - [X] (N/A) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [X] (N/A) Production readiness review completed
+- [X] (N/A) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The Kubernetes project has created and runs the `registry.k8s.io` image registry which is fully controlled by the community.
+This registry has been [GA for several months](https://kubernetes.io/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/) now and we need to freeze the old image registry.
+
+## Motivation
+
+Running public image registries is very expensive and eats up a significant chunk of the project's Infrastructure budget. We built `registry.k8s.io` image registry to serve images from various origins around the world depending on the location of the user. For example, an a kops Kubernetes cluster in eu-west-1 can pull images from an AWS S3 bucket in the same region which is very fast and more importantly very cheap for the Kubernetes project.
+
+There was a plan to redirect `k8s.gcr.io` to `registry.k8s.io` but it didn't [work out](https://kubernetes.slack.com/archives/CCK68P2Q2/p1666725317568709) so we backported the image registry defaults to 1.25, 1.24, 1.23, and 1.22 so all the patch releases from December 2022 will using the new registry by default.
+
+We are currently exceeding our budget as it will take quite a while for end users to upgrade Kubernetes to v1.25 so we want to incentivise our end users to move to the new registry as fast as possible by freezing the registry by 1.27. This would mean that all subsequent image releases will not be available on the old registry.
+
+### Goals
+
+Freeze `k8s.gcr.io` image registry and push all new images and tags exclusively to the `registry.k8s.io` image registry.
+
+### Non-Goals
+
+- `registry.k8s.io` internal implementations details. That is handled separately by sig-k8s-infra.
+
+## Proposal
+
+Freeze the `k8s.gcr.io` image by not pushing any new digests or tags after 1.27 release. The 1.27 release itself won't be available on `k8s.gcr.io`.
```

Comment:
This wording probably could be more explicit: 

The plan here is to not push any new digests for 1.24-1.26 after the 1.27 release **AND** we will not push 1.27 to that registry at all. Patch releases for the currently supported releases will land there until the 1.27 release occurs. Is that correct?

Reply:
I agree.

I'm proposing that on 01/04/2023:
- `k8s.gcr.io` is frozen and no new images will be published by any subproject.
- last 1.23 release on `k8s.gcr.io` will be 1.23.18 (goes EoL before the freeze)
- last 1.24 release on `k8s.gcr.io` will be 1.24.12
- last 1.25 release on `k8s.gcr.io` will be 1.25.8
- last 1.26 release on `k8s.gcr.io` will be 1.26.3
- 1.27.0 will not be available `k8s.gcr.io`

Reply:
1.24 had registry support backported for a while so it is reasonable to ask those users to change registries for future patch releases

Reply:
I have added this timeline to the KEP.